bin/ream/src/main.rs

```diff
@@ -30,17 +31,25 @@ async fn main() {
         Commands::Node(config) => {
             info!("starting up...");
 
-            let bootnodes = Bootnodes::new(config.network.network);
+            let server_config = ServerConfig::from_args(
+                config.http_port,
+                config.http_address,
+                config.http_allow_origin,
+            );
+
+            let http_future = start_server(config.network.genesis.clone().into(), server_config);
```

Comment:
```suggestion
            let http_future = start_server(config.network.clone(), server_config);
```

I think it would be appropriate to pass a Arc of network into the server